openlibrary/core/lending.py

```diff
@@ -111,6 +112,22 @@ def compose_ia_url(limit=None, page=1, subject=None, query=None, sorts=None, adv
     if subject:
         q += " AND openlibrary_subject:" + subject
 
+    if work_id:
+        work = web.ctx.site.get(work_id)
+        if work and _type.lower() in ["authors", "subjects"]:
```

Comment:
Can we bail/throw an error if this happens? That way we can get rid of a level of nesting :)